tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ImdsV2TestStoreCleaner.cs

```diff
@@ -0,0 +1,238 @@
+
+                            if (looksFakeEp || expiredByPolicy)
+                            {
+                                try
+                                { store.Remove(c); }
+                                catch { /* ignore */ }
+                                continue;
+                            }
+                        }
+
+                        // Case 2: Subject-based cleanup for test certs
+                        var cn = GetCn(c);
+                        var dc = GetDc(c);
+
+                        if (MatchesSubjectTrash(cn, dc))
+                        {
+                            try
+                            { store.Remove(c); }
+                            catch { /* ignore */ }
+                            continue;
+                        }
+                    }
+                    finally
+                    {
+                        c.Dispose();
+                    }
+                }
+            }
+            catch
+            {
+                // best-effort
+            }
+        }
+
+        /// <summary>
+        /// Remove all entries for a specific alias (cache key) based on FriendlyName.
+        /// Best-effort, no-throw.
+        /// </summary>
+        public static void RemoveAlias(string alias)
+        {
+            if (!IsWindows || string.IsNullOrWhiteSpace(alias))
+                return;
+
+            try
+            {
+                using var store = new X509Store(StoreName.My, StoreLocation.CurrentUser);
+                store.Open(OpenFlags.ReadWrite);
+
+                // Snapshot for safe removal
+                X509Certificate2[] items;
+                try
+                {
+                    items = new X509Certificate2[store.Certificates.Count];
+                    store.Certificates.CopyTo(items, 0);
+                }
+                catch
+                {
+                    items = store.Certificates.Cast<X509Certificate2>().ToArray();
+                }
+
+                foreach (var c in items)
+                {
+                    try
+                    {
+                        var fn = c.FriendlyName ?? string.Empty;
+                        if (fn.StartsWith(FriendlyPrefix, StringComparison.Ordinal) &&
+                            fn.Contains("alias="))
+                        {
+                            try
+                            { store.Remove(c); }
+                            catch { /* ignore */ }
+                        }
+                    }
+                    finally
+                    {
+                        c.Dispose();
+                    }
+                }
+            }
+            catch
+            {
+                // best-effort
+            }
+        }
+
+        // ---------------- helpers ----------------
+
+        private static bool MatchesSubjectTrash(string cn, string dc)
```

Comment:
Can you add XML docs for the helper methods?

Reply:
the method names themselves are self-explanatory, but still added.